apps/src/gamelab/ErrorDialogStack.jsx

```diff
@@ -3,39 +3,117 @@ import React, {PropTypes} from 'react';
 import * as actions from './errorDialogStackModule';
 import {connect} from 'react-redux';
 import BaseDialog from '../templates/BaseDialog.jsx';
+import {pegasus} from '@cdo/apps/lib/util/urlHelpers';
+import gamelabMsg from '@cdo/gamelab/locale';
+import msg from '@cdo/locale';
+import Button from '@cdo/apps/templates/Button';
+import DialogFooter from '@cdo/apps/templates/teacherDashboard/DialogFooter';
+import * as animationActions from './animationListModule';
+import firehoseClient from '@cdo/apps/lib/util/firehose';
 
 /**
  * Renders error dialogs in sequence, given a stack of errors.
  */
 class ErrorDialogStack extends React.Component {
   static propTypes = {
+    //From redux
     errors: PropTypes.arrayOf(PropTypes.object).isRequired,
-    dismissError: PropTypes.func.isRequired
+    dismissError: PropTypes.func.isRequired,
+    deleteAnimation: PropTypes.func,
+    animationList: PropTypes.object
   };
 
+  handleDeleteChoice(key){
+    // Log data about when this scenario occurs
+    firehoseClient.putRecord(
+     'analysis-events',
+        {
+          study: 'animation_no_load',
+          study_group: 'animation_no_load_with_buttons',
+          event: 'delete_selected',
+          data_json: JSON.stringify({'version': this.props.animationList.propsByKey[key].version,
+            'animationName': this.props.animationList.propsByKey[key].name})
+        }
+    );
+    this.props.deleteAnimation(key);
+    this.props.dismissError();
+  }
+
+  handleReloadChoice(key){
+    // Log data about when this scenario occurs
+    firehoseClient.putRecord(
+     'analysis-events',
+        {
+          study: 'animation_no_load',
+          study_group: 'animation_no_load_with_buttons',
+          event: 'reload_selected',
+          data_json: JSON.stringify({'version': this.props.animationList.propsByKey[key].version,
+            'animationName': this.props.animationList.propsByKey[key].name})
+        }
+    );
+    location.reload();
+  }
+
   render() {
     if (this.props.errors.length === 0) {
       return null;
     }
 
+    const error = this.props.errors[0];
+    const animationName =  (this.props.animationList && this.props.animationList.propsByKey[error.error_cause]) ?
```

Comment:
spacing should be `animationName = (...`. Also, very long line should either be broken after `?` and `:`, or (even better) you could extract `(this.props.animationList && this.props.animationList.propsByKey[error.error_cause])` into a local variable.